Source/Public/Set-VSTeamVariableGroupVariable.ps1

```diff
@@ -0,0 +1,58 @@
+function Set-VSTeamVariableGroupVariable {
+   [CmdletBinding(SupportsShouldProcess = $true, ConfirmImpact = "Medium",
+    HelpUri='https://methodsandpractices.github.io/vsteam-docs/docs/modules/vsteam/commands/Set-VSTeamVariableGroupVariable')]
+   param(
+      [Parameter(Mandatory = $true, ValueFromPipelineByPropertyName = $true)]
+      [string] $GroupName,
+
+      [Parameter(Mandatory = $true)]
+      [string] $Name,
+
+      [string] $Value,
+
+      [switch] $Force,
+
+      [Parameter(Mandatory = $true, ValueFromPipelineByPropertyName = $true)]
+      [vsteam_lib.ProjectValidateAttribute($false)]
+      [ArgumentCompleter([vsteam_lib.ProjectCompleter])]
+      [string] $ProjectName
+   )
+
+   process {
+      if ($Force -or $pscmdlet.ShouldProcess("$GroupName.$Name", "Set Variable Value")) {
+         try {
+            $gl = Get-VSTeamVariableGroup -ProjectName $ProjectName
+            # Exception if a group with a name is not found is not particularly descriptive
+            $g = $gl | Where-Object -FilterScript {$_.Name -eq $GroupName}
+            if($g) {
+               if($g.variables | Get-Member -Name $Name) {
+                  if(($g.variables.$Name | Get-Member -Name isSecret) -and $g.variables.$Name.isSecret) {
+                     Write-Error "The variable $Name is a secret one. Updating secret variables is not currently supported."
+                     return
+                  }
+                  else {
+                      $g.variables.$Name.value = $Value
+                  }
+               }
+               else {
+                  $Vars = @{}
+                  foreach($v in $g.variables.PSObject.Properties) {
+                     $Vars[$v.Name] = $v.Value
+                  }
+                  $Vars[$Name] = @{value=$Value}
+                  $g.variables = New-Object PSObject -Property $Vars
```

Comment:
Instead of recreating the whole object, rather use the Add-Member cmdlet to add a new object to the existing varaibles:

`Add-Member -InputObject $g.variables -MemberType NoteProperty -Name test -Value ([pscustomobject]@{value='val1'})`

Easier to understand and less code.